test/ERC20CosmosCoinAmountConversionTest.t.sol

```diff
@@ -0,0 +1,436 @@
+
+        // Expected values for 18 decimals
+        uint64 expectedConvertedAmount = 1_000_000_111; // 1000,000,111 Cosmos coins
+        uint256 expectedRemainder = 1;
+
+        // Assertions
+        assertEq(decimals, customMockERC20Metadata.decimals(), "Decimals mismatch");
+        assertEq(convertedAmount, expectedConvertedAmount, "Converted amount mismatch");
+        assertEq(remainder, expectedRemainder, "Remainder mismatch");
+    }
+
+    function testConvertCosmosCoinAmountToERC20_SameDecimals() public {
+        MockERC20Metadata customMockERC20Metadata = new MockERC20Metadata(6);
+        uint64 cosmosAmount = 1_000_000; // 1 Cosmos coin
+        uint256 convertedAmount = ERC20CosmosCoinAmountConversion._convertCosmosCoinAmountToERC20(
+            cosmosAmount, address(customMockERC20Metadata)
+        );
+        uint256 expectedAmount = 1_000_000; // 1 ERC20 token with 6 decimals
+
+        assertEq(convertedAmount, expectedAmount, "Conversion mismatch for same decimals");
+    }
+
+    function testConvertCosmosCoinAmountToERC20_1() public {
+        MockERC20Metadata customMockERC20Metadata = new MockERC20Metadata(18);
+        uint64 cosmosAmount = 1_000_000; // 1 Cosmos coin
+        uint256 convertedAmount = ERC20CosmosCoinAmountConversion._convertCosmosCoinAmountToERC20(
+            cosmosAmount, address(customMockERC20Metadata)
+        );
+        uint256 expectedAmount = 1_000_000_000_000_000_000; // 1 ERC20 token with 18 decimals
+
+        assertEq(convertedAmount, expectedAmount, "Conversion mismatch for higher decimals");
+    }
+
+    function testConvertCosmosCoinAmountToERC20_2() public {
+        MockERC20Metadata customMockERC20Metadata = new MockERC20Metadata(7);
+        uint64 cosmosAmount = 1_000_000; // 1 Cosmos coin
+        uint256 convertedAmount = ERC20CosmosCoinAmountConversion._convertCosmosCoinAmountToERC20(
+            cosmosAmount, address(customMockERC20Metadata)
+        );
+        uint256 expectedAmount = 10_000_000; // 1 ERC20 token with 9 decimals
+
+        assertEq(convertedAmount, expectedAmount, "Conversion mismatch for 9 decimals");
+    }
+
+    function testConvertCosmosCoinAmountToERC20_3() public {
+        MockERC20Metadata customMockERC20Metadata = new MockERC20Metadata(77);
+        uint64 cosmosAmount = 1_000_000; // 1 Cosmos coin
+        uint256 convertedAmount = ERC20CosmosCoinAmountConversion._convertCosmosCoinAmountToERC20(
+            cosmosAmount, address(customMockERC20Metadata)
+        );
+        uint256 expectedAmount =
+            100_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000_000; // 1
+            // ERC20 token with 18 decimals
+
+        assertEq(convertedAmount, expectedAmount, "Conversion mismatch for higher decimals");
+    }
+
+    function testConvertCosmosCoinAmountToERC20_LessThanSixDecimals() public {
```

Comment:
I think it would be good to try to cover a few more failure scenarios: ideally triggering all the reverts we have if possible :)

Reply:
Addressed [here](https://github.com/srdtrk/solidity-ibc-eureka/pull/19/commits/785a18c932a601a05ba7e801410705d05001d33a)

Reply:
I think we can add more failure scenarios. But we don't need to hit everything as long as we add a `TODO` for this and address it in the future